### PR TITLE
Created a part of UserService test

### PR DESCRIPTION
**Pull request by danilchican**

### CODE REVIEW

File: final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @InjectMocks
+    private UserService userService;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void getUserByEmailTest() throws Exception {
```
Comment (ghost) [inline final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java]: I'd propose folllow one of the listed naming conventions for unit tests: https://dzone.com/articles/7-popular-unit-test-naming
My personal preference - options 1 and 2 from the list.

File: final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        User expected = new User();
```
Comment (ghost) [inline final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java]: Use mocked instance instead. There are two approaches:
1) using annotations
@Mock
private User user;
2) using Mockito.mock method
User user = mock(User.class)

File: final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Collection;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = { "classpath:spring.xml" })
```
Comment (ghost) [inline final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java]: I'd propose to revert to more classical approach using @ RunWith(MockitoJUnitRunner.class).

Comment (danilchican) [reply final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java]: I tried to use it, but it was failed. I'll try again.

File: final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        User actual = userService.getUserByEmail("danilchican@mail.ru");
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void getUserByEmailFirstFailureScenarioTest() throws Exception {
+        when(userDAO.getUserByEmail(anyString())).thenReturn(Optional.empty());
+        User actual = userService.getUserByEmail("danilchican@mail.ru");
+
+        assertThat(actual, is(nullValue()));
```
Comment (aryadnabirchanka) [inline final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java]: Swap actual and expected results. Expected result should be first.

File: final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    @Mock
+    private UserDAO userDAO;
+
+    @Autowired
+    @InjectMocks
+    private UserService userService;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
```
Comment (ghost) [inline final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java]: https://stackoverflow.com/a/10812752 - to see the difference between 
@RunWith(MockitoJUnitRunner.class) and MockitoAnnotations.initMocks(this);

File: final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    @Test
+    public void getUserByEmail_ReturnsOptionalUser_WhenUserWithSuchEmailExists() throws Exception {
+        when(userDAO.getUserByEmail(anyString())).thenReturn(Optional.of(user));
+
+        User actual = userService.getUserByEmail("danilchican@mail.ru");
+        assertEquals(user, actual);
+    }
+
+    @Test
+    public void getUserByEmail_ReturnsEmpty_WhenUserWithSuchEmailDoesntExists() throws Exception {
```
Comment (ghost) [inline final/src/test/java/ua/epam/spring/hometask/service/UserServiceImplTest.java:41]: TYPO: ...DoesntExist

**Review by ghost — APPROVED** [review]

**Pull request merged**